lumicks/pylake/tests/test_imaging_camera/test_correlated_stack.py

```diff
@@ -63,6 +60,44 @@ def test_correlated_stack(shape):
     np.testing.assert_allclose([x.start for x in stack[2]], [30])
 
 
+@pytest.mark.parametrize("shape", [(3,3), (5,4,3)])
+def test_slicing(shape):
+    image = [np.random.poisson(10, size=shape) for _ in range(10)]
+    times = [[f"{j}", f"{j+8}"] for j in range(10, 1100, 10)]
+    fake_tiff = TiffStack(MockTiffFile(data=image, times=times), align_requested=False)
+    stack0 = CorrelatedStack.from_dataset(fake_tiff)
+
+    def compare_frames(original_frames, new_stack):
+        for new_frame_index, index in enumerate(original_frames):
+            frame = stack0._get_frame(index).data
+            new_frame = new_stack._get_frame(new_frame_index).data
+            np.testing.assert_equal(frame, new_frame)
+
+    compare_frames([0], stack0[0])  # first frame
+    compare_frames([9], stack0[-1])  # last frame
+    compare_frames([3], stack0[3])  # single frame
+    compare_frames([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], stack0[:])  # all frames
+    compare_frames([3, 4, 5], stack0[3:6])  # normal slice
+    compare_frames([0, 1, 2], stack0[:3]) #  from beginning
+    compare_frames([6, 7, 8, 9], stack0[6:]) #  until end
+    compare_frames([0, 1, 2, 3, 4, 5], stack0[:-4]) #  until negative index
+    compare_frames([5, 6, 7], stack0[5:-2]) #  mixed sign indices
+    compare_frames([6, 7], stack0[-4:-2]) #  negative indices slice
+    compare_frames([3, 4], stack0[2:6][1:3]) #  iterative slicing
+
```

Comment:
I like that we now have all these slicing tests.

I'd suggest three extra cases:
```python
    compare_frames([1, 2, 3, 4, 5, 6, 7, 8, 9], stack0[1:1000])  # test clamping past the end
    compare_frames([0, 1, 2, 3, 4, 5, 6, 7, 8], stack0[-1000:10])  # test clamping past the beginning
    compare_frames([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], stack0[-1000:1000])  # test clamping both dirs
```